### ISSUE TITLE
We should add an FAQ for the #minishift channel and reference it in the channel topic

### ISSUE THREAD

**Issue by veillard** [minishift/minishift#547]
Very common practice, miminize the time spent answering random questions

**Comment by Preeticp** [comment]
We could use this issue to list some of the frequently asked questions we want to include in the docs. @hferentschik , @gbraad, @LalatenduMohanty, others, do we have any questions and related answers we can begin listing for this section. 
My 2 cents, both Trouble shooting and FAQ are pretty closely related, hence could reside in the Trouble shooting file itself, with probably a separate section for FAQ. Maybe we could rename it to Trouble shooting and FAQ.

**Comment by thatdocslady** [comment]
My 2 korunas :) if the FAQ is meant as a quick help guide for common issues, then it's basically the same as the troubleshooting guide, and we can probably add more common issues and restructure the existing doc. If the FAQ is meant for more general Q&A (for example, "what's the difference between minishift and openshift" and similar questions) then it should be added to the getting started guide as "here are questions you might have before you get started".

Also just to clarify, do you mean that you want to add a link to this document to the IRC channel topic, or actually have some sort of bot that answers questions in the channel?

**Comment by budhrg** [comment]
For me, FAQ means the list of short questions and answers list which comes on users's mind like below

- What is Minishift ? (I know it is there in README but we can list here too)
- Is it same as OpenShift?
- What is difference between Minishift and OpenShift?
- Can I run in multiple Operating Systems?
- Can we use Persistent Volume ? (No right now is the answer and link to limitation)
- ....

many more such short questions and answers

Troubleshooting on the other hand is more description sort of questions and need more detailed explaination.

+1 for both.

**Comment by LalatenduMohanty** [comment]
> f the FAQ is meant for more general Q&A (for example, "what's the difference between minishift and openshift" and similar questions) then it should be added to the getting started guide as "here are questions you might have before you get started".
> 
> Also just to clarify, do you mean that you want to add a link to this document to the IRC channel topic.

+1 good idea. We can add this channel topic .

**Comment by gbraad** [comment]
@thatdocslady actiually set a channel topic this morning. An actually single endpoint is easier than just directing people to 'somewhere on minishift.io'

**Comment by thatdocslady** [comment]
@gbraad of course, we should have a single link to the entry point. @LalatenduMohanty if this is indeed what @veillard meant, then I can add a topic to the getting started section and each Q&A will act like a gateway to a different section of the documentation if relevant (for example, "what do I do if I run into a problem? -Here is a link to troubleshooting", and "how do I find out the default driver for me OS").

WDYT? If anyone has suggestions for Q&A topics please write them here and I'll compile them.

**Comment by Preeticp** [comment]
Adding a question which could be added to the FAQ: how do you login to the web console of OpenShift as system:admin?
Answer: this is not possible. 
@hferentschik is there anything else which could be added to this answer.

**Comment by budhrg** [comment]
With the question asked by one of user [here](https://blog.openshift.com/goodbye-openshift-all-in-one-vm-hello-minishift/#comment-3179923296), it is definitely a requirement to document somwhere `What Minishift brings more than "oc cluster up"?`.  Might a dedicated section in README instead of FAQ will highlight more.

**Comment by gbraad** [comment]
Of course, it is understandable some explanation is in place, but it is
also like answering "why would you run a build in a clean environment",
"why would you use containers?". Part of these questions are related to the
general understanding.of the technology they are using.

**Comment by agajdosi** [comment]
@LalatenduMohanty We still do not have any kind of FAQ and it seems to me that points of @veillard are still valid. I think we could add it for 1.8.0 or 1.9.0 milestones, we could bring our ideas for Q&As collectively and later it could be not so hard to compile them into one file, wdyt?

**Comment by gbraad** [comment]
I suggest we enable the GitHub Wiki for this project and start adding questions/answers. I do not think using document generation and a publishing step is really helpful here.

Note: this can still be done, as GitHub wiki is just a Git repo with documents stored in markdown format (using Gollum as backend).

**Comment by LalatenduMohanty** [comment]
> we could bring our ideas for Q&As collectively and later it could be not so hard to compile them into one file, wdyt?

@agajdosi Agree. 


> I suggest we enable the GitHub Wiki for this project and start adding questions/answers. I do not think using document generation and a publishing step is really helpful here.

@gbraad My concern around the wiki is it can not be version controlled and also it does not go through review.

**Comment by hferentschik** [comment]
So what do we even want to add there? Is this for Minishift specific issues? For the Minishift IRC channel? I am not really sure what we really try to solve/address here. What's about the FAQ as part of the docs? 

Regarding the Wiki, I vote against it. In my experience you end up with stale information very quickly and it adds yes another place you have to watch and try to keep up to date.

**Comment by hferentschik** [comment]
IMO this issue can be closed.

**Comment by LalatenduMohanty** [comment]
> What's about the FAQ as part of the docs?

@hferentschik  The idea is to add FAQ to the documentation and the put the link in the IRC channel topic. 

We should first create the list of questions which we should be part of FAQ and that would give a lot of context around how this will look and how much useful it will be to users.

**Comment by budhrg** [comment]
May be we should take reference from here https://istio.io/faq/. 

WDYT @veillard @LalatenduMohanty @gbraad @praveenkumar @hferentschik

**Comment by LalatenduMohanty** [comment]
@robin-owen What is your suggestion on this?

**Comment by robin-owen** [comment]
@LalatenduMohanty Looking back over the discussion here, I find that I agree with much of what @thatdocslady said. Personally, I think using a FAQ as a gateway to various areas of the documentation is a good idea.

As a secondary note, if we encounter entries to the FAQ which have no related section in the documentation, that's a significant sign that something is missing in the docs. In that case, I would greatly prefer having the information added to the documentation where appropriate and linked to from the FAQ.

I also think that the Istio FAQ is a decent reference, but personally, I'd like to keep extra information in the FAQ to a minimum. I'd like to see each entry with minimal writing that will require maintenance outside of the official documentation.

**Comment by gbraad** [comment]
@robin-owen what can be done to move this forward?
Should we use a general documentation location or the wiki ((we haven't used ever?).

**Comment by robin-owen** [comment]
@gbraad I don't think that we should use the wiki. Having two locations for common information (the docs and the wiki) would be confusing in my opinion.

On that note, I do think that a FAQ deserves its own location/section in the documentation. It doesn't fit well into any of the currently existing categories, and could be useful as an entrypoint to the information available in the overall documentation.

To help us move forward on this, we should discuss common questions and issues that relatively new users may encounter. For example, we can discuss hypervisor setup, setting persistent configuration options, using add-ons, and so on. If we can "touch" each section of the documentation in a FAQ, that would be ideal -- and as said previously, if we come across questions which aren't adequately covered in our documentation already, that's a fantastic sign that we need to expand the relevant area(s) of the documentation.
